# TASK-3

We are ready to run the CAPM analysis after all these preparation process and the interpretation is left to the `TASK-3.`

In [1]:
import statsmodels.api as sm
import pandas as pd

In [2]:
returns = pd.read_csv('returns.csv')
returns = returns.set_index('Date')

In [3]:
y_aapl = returns['excess_return_AAPL']
X = returns['excess_return_GSPC']

In [4]:
X = sm.add_constant(X, prepend=False)
CAPM_AAPL = sm.OLS(y_aapl, X)
CAPM_results_AAPL = CAPM_AAPL.fit()

In [5]:
CAPM_results_AAPL.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:     excess_return_AAPL   R-squared:                       0.442
Model:                            OLS   Adj. R-squared:                  0.432
Method:                 Least Squares   F-statistic:                     46.73
Date:                Thu, 05 Aug 2021   Prob (F-statistic):           5.13e-09
Time:                        17:13:26   Log-Likelihood:                 167.80
No. Observations:                  61   AIC:                            -331.6
Df Residuals:                      59   BIC:                            -327.4
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
======================================================================================
                         coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------
excess_return_GSPC     1.3891      0.203      6.836      0.000       0.982       1.796
const                 -0.0023      0.002     -1.126      0.265      -0.006       0.002
==============================================================================
Omnibus:                        2.777   Durbin-Watson:                   2.000
Prob(Omnibus):                  0.249   Jarque-Bera (JB):                2.142
Skew:                          -0.174   Prob(JB):                        0.343
Kurtosis:                       3.849   Cond. No.                         101.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [6]:
y_IBM=returns['excess_return_IBM']

In [7]:
X = sm.add_constant(X, prepend=False)
CAPM_IBM= sm.OLS(y_IBM, X)
CAPM_results_IBM = CAPM_IBM.fit()

In [8]:
CAPM_results_IBM.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:      excess_return_IBM   R-squared:                       0.066
Model:                            OLS   Adj. R-squared:                  0.050
Method:                 Least Squares   F-statistic:                     4.187
Date:                Thu, 05 Aug 2021   Prob (F-statistic):             0.0452
Time:                        17:13:26   Log-Likelihood:                 159.59
No. Observations:                  61   AIC:                            -315.2
Df Residuals:                      59   BIC:                            -311.0
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
======================================================================================
                         coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------
excess_return_GSPC     0.4757      0.232      2.046      0.045       0.010       0.941
const                  0.0003      0.002      0.151      0.881      -0.004       0.005
==============================================================================
Omnibus:                       62.009   Durbin-Watson:                   2.274
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              531.745
Skew:                          -2.671   Prob(JB):                    3.41e-116
Kurtosis:                      16.442   Cond. No.                         101.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [9]:
y_MSFT=returns['excess_return_MSFT']

In [10]:
X = sm.add_constant(X, prepend=False)
CAPM_MSFT= sm.OLS(y_MSFT, X)
CAPM_results_MSFT = CAPM_MSFT.fit()

In [11]:
CAPM_results_MSFT.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:     excess_return_MSFT   R-squared:                       0.445
Model:                            OLS   Adj. R-squared:                  0.435
Method:                 Least Squares   F-statistic:                     47.26
Date:                Thu, 05 Aug 2021   Prob (F-statistic):           4.41e-09
Time:                        17:13:26   Log-Likelihood:                 184.67
No. Observations:                  61   AIC:                            -365.3
Df Residuals:                      59   BIC:                            -361.1
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
======================================================================================
                         coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------
excess_return_GSPC     1.0595      0.154      6.875      0.000       0.751       1.368
const               8.649e-05      0.002      0.057      0.955      -0.003       0.003
==============================================================================
Omnibus:                        0.098   Durbin-Watson:                   1.988
Prob(Omnibus):                  0.952   Jarque-Bera (JB):                0.160
Skew:                           0.088   Prob(JB):                        0.923
Kurtosis:                       2.823   Cond. No.                         101.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [12]:
y_INTC=returns['excess_return_INTC']

In [13]:
X = sm.add_constant(X, prepend=False)
CAPM_INTC= sm.OLS(y_INTC, X)
CAPM_results_INTC = CAPM_INTC.fit()

In [14]:
CAPM_results_INTC.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:     excess_return_INTC   R-squared:                       0.382
Model:                            OLS   Adj. R-squared:                  0.371
Method:                 Least Squares   F-statistic:                     36.41
Date:                Thu, 05 Aug 2021   Prob (F-statistic):           1.14e-07
Time:                        17:13:26   Log-Likelihood:                 146.81
No. Observations:                  61   AIC:                            -289.6
Df Residuals:                      59   BIC:                            -285.4
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
======================================================================================
                         coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------
excess_return_GSPC     1.7298      0.287      6.034      0.000       1.156       2.304
const                  0.0032      0.003      1.140      0.259      -0.002       0.009
==============================================================================
Omnibus:                       14.511   Durbin-Watson:                   1.934
Prob(Omnibus):                  0.001   Jarque-Bera (JB):               56.109
Skew:                          -0.257   Prob(JB):                     6.55e-13
Kurtosis:                       7.670   Cond. No.                         101.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

First, we run the CAPM analysis for `Apple` and it turns out that Apple is more volatile than the S&P-500 as we have a beta coefficient of 1.3891. This estimated beta coefficient is statistically significant at 1% level since we have a t-value of 6.847 (or p-value of 0.0000). However, the intercept term is not statistically significant. Therefore, interpretation of the intercept term, which indicates the performance of the technical analysis, does not make sense.

When it comes to `IBM`, the result is quite similar in terms of slope coefficien $\beta$ and intercept term. The beta coefficient of 0.4757 implies that there is a correlation between excess market return and excess return of the IBM. More specifically, 1-unit increase in S&P-500 leads to an increase in IBM return by nearly 0.4757. The result also suggests that the IBM is less risky than market index.  Again, the intercept term is not statistically significant at conventional confidence interval.

`Microsoft` does tell us a similar story with positive and statistically significant beta coefficient and statistically insignificant intercept term. The magnitude of beta coefficient is 1.0595 indicating that 1-unit increase in S&P-500 results in an increase of 1.0595 in Microsoft's return.

Finally, INTEL has the highest estimated coefficient, implying that INTEL has the most risky company in the portfolio.

In summary, given the result obtained via CAPM, the most and least risky or volatile stocks are INTEL and IBM, respectively.